drools/src/main/java/com/datatorrent/drools/operator/StatelessDroolsOperator.java

```diff
@@ -140,4 +164,124 @@ public StatelessKieSession getKieSession()
   {
     return kieSession;
   }
+
+  private class RulesFiredListener implements AgendaEventListener
+  {
+
+    RulesFiredListener()
+    {
+
+    }
+
+    @Override
+    public void matchCreated(MatchCreatedEvent matchCreatedEvent)
+    {
+
+    }
+
+    @Override
+    public void matchCancelled(MatchCancelledEvent matchCancelledEvent)
+    {
+
+    }
+
+    @Override
+    public void beforeMatchFired(BeforeMatchFiredEvent event)
+    {
+
+    }
+
+    /**
+     * This function will be called automatically after firing a Rule.
+     * Modifies the map factsAndFiredRules and ruleCount.
+     * @param afterMatchFiredEvent the event which triggered this call
+     */
+    @Override
+    public void afterMatchFired(AfterMatchFiredEvent afterMatchFiredEvent)
+    {
+      Rule matchedRule = afterMatchFiredEvent.getMatch().getRule();
+      for (Object matchedObject : afterMatchFiredEvent.getMatch().getObjects()) {
```

Comment:
this way its consistent with the api call, matchedFact should actually be the fact handle returned by getFact()